WIP2 - Upgrade to rails 6.1 [from fork: replaced]

ianheggie-oaf opened the pull request:
Upgrades morph to rails 6.1

Builds on
* https://github.com/openaustralia/morph/pull/1367

Related to:
* https://github.com/openaustralia/morph/issues/1354

ianheggie-oaf committed:
Add html and console coverage format

ianheggie-oaf committed:
Add convienience bash aliases

ianheggie-oaf committed:
Add db:stats rake task

ianheggie-oaf committed:
Update to rails 6.0.6.1 in Gemfile, including related gems

ianheggie-oaf committed:
Update to rails 6.0.6.1 in Gemfile, including related gems

ianheggie-oaf committed:
Update sorbet gem type defs

ianheggie-oaf committed:
Update sorbet gem type defs

ianheggie-oaf committed:
Move require 'csv' to controller that needs it

ianheggie-oaf committed:
Move require 'csv' to controller that needs it

ianheggie-oaf committed:
Fix LoggerThreadSafeLevel not defined because it wasn't loaded by rails

ianheggie-oaf committed:
Fix LoggerThreadSafeLevel not defined because it wasn't loaded by rails

ianheggie-oaf committed:
app:update updates, includes some cleanup

ianheggie-oaf committed:
app:update updates, includes some cleanup

ianheggie-oaf committed:
Fix database.yml can no longer use macros to be DRY

ianheggie-oaf committed:
Fix database.yml can no longer use macros to be DRY

ianheggie-oaf committed:
Committed db/schema.rb from production

ianheggie-oaf committed:
Fix create scraper migration so it matches actual production

ianheggie-oaf committed:
Fix create scraper migration so it matches actual production

ianheggie-oaf committed:
Committed db/schema.rb from production

ianheggie-oaf committed:
Update database.yml.example to match production encoding/collation and use common 2 space indentation

ianheggie-oaf committed:
Update database.yml.example to match production encoding/collation and use common 2 space indentation

ianheggie-oaf committed:
Add app:update migration, db/schema.rb reflects reordering of 2 columns

ianheggie-oaf committed:
Add app:update migration, db/schema.rb reflects reordering of 2 columns

ianheggie-oaf committed:
Explain extra test options in Makefile

ianheggie-oaf committed:
Explain extra test options in Makefile

ianheggie-oaf committed:
specify is uniqueness is case sensitive or not to fix deprecation warning

ianheggie-oaf committed:
specify is uniqueness is case sensitive or not to fix deprecation warning

ianheggie-oaf committed:
Fix content_type deprecation warning in tests

ianheggie-oaf committed:
Fix content_type deprecation warning in tests

ianheggie-oaf committed:
Fix deprecation about autoload classes in initializers conflict

ianheggie-oaf committed:
Fix deprecation about autoload classes in initializers conflict

ianheggie-oaf committed:
don't reference active_storage if not used

ianheggie-oaf committed:
don't reference active_storage if not used

ianheggie-oaf committed:
fix DEPRECATION WARNING: Initialization autoloaded the constants ActionText::ContentHelper and ActionText::TagHelper
by excluding unessisary parts of rails. Also reduces memory required

ianheggie-oaf committed:
fix DEPRECATION WARNING: Initialization autoloaded the constants ActionText::ContentHelper and ActionText::TagHelper
by excluding unessisary parts of rails. Also reduces memory required

ianheggie-oaf committed:
Allow exposure of development server via ngrok

ianheggie-oaf committed:
Allow exposure of development server via ngrok

ianheggie-oaf committed:
Set minimum coverage

ianheggie-oaf committed:
Set minimum coverage

ianheggie-oaf committed:
Enable 5.2 framework defaults that had not previously been enabled

ianheggie-oaf committed:
Enable 5.2 framework defaults that had not previously been enabled

ianheggie-oaf committed:
Update to rails 6.0 defaults

ianheggie-oaf committed:
Update to rails 6.0 defaults

ianheggie-oaf committed:
Update rails gem

ianheggie-oaf committed:
Update rails gem

ianheggie-oaf committed:
Regenerate Sorbet RBI files for gems

ianheggie-oaf committed:
Regenerate Sorbet RBI files for gems

ianheggie-oaf committed:
Updated by rails app:update command

ianheggie-oaf committed:
Updated by rails app:update command

ianheggie-oaf committed:
Ran bin/yarn as per bin/setup

ianheggie-oaf committed:
Ran bin/yarn as per bin/setup

ianheggie-oaf committed:
disable spring as it was causing Rails::Application is abstract, you cannot instantiate it directly in railties gem

ianheggie-oaf committed:
disable spring as it was causing Rails::Application is abstract, you cannot instantiate it directly in railties gem

ianheggie-oaf committed:
Fix change in helper in mailers

ianheggie-oaf committed:
Fix change in helper in mailers

ianheggie-oaf committed:
Update errors.keys to replacement method

ianheggie-oaf committed:
Update errors.keys to replacement method

ianheggie-oaf committed:
Update to rails 6.1 config

ianheggie-oaf committed:
Update to rails 6.1 config

ianheggie-oaf committed:
run `rubocop -a` update to pass github lint

ianheggie-oaf committed:
run `rubocop -a` update to pass github lint

ianheggie-oaf committed:
Reduce test coverage required if docker tests are not run

ianheggie-oaf committed:
Reduce test coverage required if docker tests are not run

Br3nda commented on app/models/webhook.rb:
```diff
@@ -6,7 +6,7 @@ class Webhook < ApplicationRecord
   belongs_to :scraper
   has_many :deliveries, class_name: "WebhookDelivery", dependent: :delete_all
-  validates :url, presence: true, url: true, uniqueness: { scope: :scraper }
+  validates :url, presence: true, url: true, uniqueness: { scope: :scraper, case_sensitive: true }
```
I think case_sensitive is the default anyway.

Br3nda commented on bin/setup:
```diff
@@ -9,24 +8,25 @@ def system!(*args)
-  # Install JavaScript dependencies if using Yarn
-  # system('bin/yarn')
+  # Install JavaScript dependencies
+  system! 'bin/yarn'
```
Are we sure we need this? Did we add yarn?

ianheggie-oaf replied:
I left it as app:update wanted, as otherwise i would be fighting with app:update on each update

Br3nda commented on bin/spring:
```diff
@@ -1,15 +1,14 @@
+    gem "spring", spring.version
     require "spring/binstub"
+  rescue Gem::LoadError
+    # Ignore when Spring is not installed.
```
did this change come with rails? Is so, all good.

could hide errors from people debugging one day

ianheggie-oaf replied:
came from rails from memory - I don't directly call bin/spring so I wouldn't have had a reason to change

Br3nda commented on db/migrate/20131223033729_create_scrapers.rb:
```diff
@@ -1,7 +1,7 @@
   def change
     create_table :scrapers do |t|
-      t.string :name, null: false
+      t.string :name, default: "", null: false
```
This is an old migration that already run, so why are we modifying it?

Br3nda commented on provisioning/group_vars/all.yml:
```diff
@@ -25,6 +25,12 @@ apt_packages:
+  - 'alias be="bundle exec"'
+  - 'alias r="bundle exec rails"'
+  - 'alias rk="bundle exec rake"'
+
```
Is that were we install on prod?  (okay, I  checked it is)
We're using `/srv/` on other deploys.

Br3nda replied:
these do look like personal aliases, but I can't see the harm in having them.

Br3nda reviewed (commented):
added comments. none are blocking comments. Happy to approve, just checking for any replies from @ianheggie-oaf  before that

ianheggie-oaf committed:
Merge pull request #1390 from openaustralia/feat/ben-ssh so I could push the change to production

Add 'benrfairless' to the list of GitHub users for SSH Access

ianheggie-oaf committed:
change from unsupported innobackupex to mysqldump and compress backups

ianheggie-oaf committed:
Increase the number of open file descriptors for duply

ianheggie-oaf committed:
Limit backup chains to one month and keep a max of 6 of them

As MAX_AGE by itself was ineffective (we had two chains, one spanning many years)

ianheggie-oaf committed:
Merge pull request #1389 from ianheggie-oaf/add_html_coverage_report

Add html coverage report to staging branch

ianheggie-oaf committed:
Merge pull request #1393 from ianheggie-oaf/fix_backup-database_script

Fix backup database script

ianheggie-oaf committed:
Add gems to support ed25519 ssh keys

# Conflicts:
#	Gemfile.lock

# Conflicts:
#	Gemfile.lock

ianheggie-oaf committed:
Gem bcrypt_pbkdf should be sorted in an alphabetical order

ianheggie-oaf committed:
Merge pull request #1396 from ianheggie-oaf/support_ed25519_ssh_keypairs

Add gems to support ed25519 ssh keys

ianheggie-oaf committed:
Merge branch 'upgrade_to_rails_61' of github.com:ianheggie-oaf/morph into upgrade_to_rails_61
